src/util/RouterUtil.js

```diff
@@ -218,14 +218,22 @@ function (Okta, OAuth2Util, Util, Enums, BrowserFeatures, Errors, ErrorCodes) {
     case 'FACTOR_REQUIRED':
     case 'FACTOR_CHALLENGE':
     case 'MFA_REQUIRED':
+      if (router.appState.get('hasFailed') && router.appState.get('challengeErrorFactor')) {
+        router.appState.get('factors').lastUsedFactor = router.appState.get('challengeErrorFactor');
+      }
       var factor = router.appState.get('factors').getDefaultFactor();
       var url = fn.createVerifyUrl(factor.get('provider'), factor.get('factorType'));
       router.navigate(url, { trigger: true });
+      router.appState.unsetFactorResultFailedProperties();
       return;
     case 'MFA_CHALLENGE':
       // Since we normally trap MFA_CHALLENGE, this will only get called on a
-      // page refresh. We need to return to MFA_REQUIRED to initialize the
+      // page refresh or when an error is returned on verification with an IdP.
+      // We need to return to MFA_REQUIRED to initialize the
       // page correctly (i.e. factors dropdown, etc)
+      var lastAuthResponse = router.appState.get('lastAuthResponse');
```

Comment:
so with what I am suggesting, this would be something like
```
if (router.appState.get('isFactorResultFailed')) {
  router.appState.set('lastChallengedFactorData', {
    factor: router.appState.get('factor'),
    errorMessage: router.appState.get('factorResultMessage'),
  });
}
```

Reply:
lastFailedChallengedFactorData

Reply:
Changing to lastFailedChallengeFactorData